src/midi/midi_mt32.cpp

```diff
@@ -457,7 +457,7 @@ void MidiHandler_mt32::Render()
 	while (keep_rendering) {
 		service->renderFloat(render_buffer.data(), FRAMES_PER_BUFFER);
 		soft_limiter.Process(render_buffer, FRAMES_PER_BUFFER, playable_buffer);
-		ring.wait_enqueue(playable_buffer);
+		ring.wait_enqueue(std::move(playable_buffer));
```

Comment:
OK, now it's definitely wrong - we can't use `playable_buffer` after moving it out! We would need to re-initialize the vector!

Reply:
> can't use playable_buffer without re-initializing

Exactly :-)  in this second commit we want prove that move into rw-queue is occurring, so inside the soft limiter, it will re-initialize this inbound play buffer and print a log message 

``` c++
(In soft limiter process function)
if (out.size() < frames) {
	out.clear(); // re-initialize
	out.resize(frames);
	DEBUG_LOG_MSG("SOFT_LIMITER: reinitialized output buffer");
}
```
In the commit, I mention this this is an interim step, and of course we wouldn't want code like this.. but it's just a proof point. 

You'll see in the 3rd commit this become an assert just to check the size, because in the 3rd commit we achieve perfect move rotation of the 8 buffers in and out of the two rw-queues.